src/cascadia/TerminalSettingsEditor/ColorSchemes.h

```diff
@@ -13,11 +13,18 @@ namespace winrt::Microsoft::Terminal::Settings::Editor::implementation
     struct ColorSchemesPageNavigationState : ColorSchemesPageNavigationStateT<ColorSchemesPageNavigationState>
     {
     public:
-        ColorSchemesPageNavigationState(const Model::GlobalAppSettings& settings) :
-            _Globals{ settings } {}
+        ColorSchemesPageNavigationState(const Model::CascadiaSettings& settings) :
+            _Globals{ settings.GlobalSettings() },
+            _Settings{ settings } {}
+
+        void RenameColorScheme(hstring oldName, hstring newName);
+        void DeleteColorScheme(hstring name);
 
```

Comment:
This is an inappropriate responsibility for the navigation state. It's supposed to be a handoff object that gets information from MainPage into ColorSchemesPage, nothing more.

Reply:
(This one still applies under the new design)
I could do all of this in the color schemes page. The navigation state would have to expose `CascadiaSettings` though. Or is there a way that I could _only_ expose `UpdateColorSchemeReferences()` without having to make an interface for that one thing?

Reply:
Ended up exposing `CascadiaSettings` entirely instead of just `GlobalAppSettings`.